### DISCUSSION TITLE
Does this protocol support English auctions?

### DISCUSSION THREAD

**Discussion by ropital** [ProjectOpenSea/seaport#662] [Q&A]
After thinking about it for a few days, I could not come up with a way to implement an English auction. As far as I have checked, I could not find an example of an English auction implementation.

I also checked `opensea-js`, but it seems to use old `WyvernProtocol` for English auctions.

If anyone knows if the English auction can be implemented with seaport-protocol or how to implement it, please let me know🙏

**Comment by ropital** [comment]
I think we need to use zone. But I'm not sure how I can implement english auction with zone🤔

**Comment by ryanio** [comment]
You can create an english auction zone, or you can set the zone to an EOA who can programmatically execute the order with the winner of the auction. Note that for the order to show up on OpenSea currently you need to use OpenSea's required zone though. You can also create an English auction through the OpenSea UI to automatically get the benefit of OpenSea completing the auction (although the threshold is 1 eth I believe for OpenSea to cover the cost of closing the auction)

**Reply by suryaAppinventiv** [reply]
> You can create an english auction zone, or you can set the zone to an EOA who can programmatically execute the order with the winner of the auction. Note that for the order to show up on OpenSea currently you need to use OpenSea's required zone though. You can also create an English auction through the OpenSea UI to automatically get the benefit of OpenSea completing the auction (although the threshold is 1 eth I believe for OpenSea to cover the cost of closing the auction)

Hi, thanks for the excellent answer you provided, but I am not able fully able to understand how we can implement an auction zone because the zone can validate a transaction if it is good to go or not, but an auction is a long process. In auctions, people do bids and withdraw bids and at the end transfer/buying happens, so how can this be implemented? can you please help and explain my doubt?

**Reply by ryanio** [reply]
The zone should be set to an EOA and that EOA can call to fulfill the order with the highest bid, as it has permission to execute the order by being listed as the zone. Does that help?
